test/source/tests/browser-unit-tests/unit-ContactStore.js

```diff
@@ -0,0 +1,349 @@
+    throw Error(`Expected to get the key fingerprint 3155F118B6E732B3638A1CE1608BCD797A23FB91 but got ${abcdefByPrimaryLongid.pubkey.id}`)
+  }
+  const [abcdefBySubkeyLongid] = await ContactStore.get(undefined, ['2D47A41943DFAFCE']);
+  if (abcdefBySubkeyLongid.email !== 'abcdef@test.com') {
+    throw Error(`Expected to get the key for abcdef@test.com by subkey longid but got ${abcdefBySubkeyLongid.email}`)
+  }
+  if (abcdefBySubkeyLongid.pubkey.id !== '3155F118B6E732B3638A1CE1608BCD797A23FB91') {
+    throw Error(`Expected to get the key fingerprint 3155F118B6E732B3638A1CE1608BCD797A23FB91 but got ${abcdefBySubkeyLongid.pubkey.id}`)
+  }
+
+  const [abcddfByPrimaryLongid] = await ContactStore.get(undefined, ['75AA44AB8930F7E9']);
+  if (abcddfByPrimaryLongid.email !== 'abcddf@test.com') {
+    throw Error(`Expected to get the key for abcddf@test.com by primary longid but got ${abcddfByPrimaryLongid.email}`)
+  }
+  if (abcddfByPrimaryLongid.pubkey.id !== '6CF53D2329C2A80828F499D375AA44AB8930F7E9') {
+    throw Error(`Expected to get the key fingerprint 6CF53D2329C2A80828F499D375AA44AB8930F7E9 but got ${abcddfByPrimaryLongid.pubkey.id}`)
+  }
+  const [abcddfBySubkeyLongid] = await ContactStore.get(undefined, ['92CFDAC7AA3A4253']);
+  if (abcddfBySubkeyLongid.email !== 'abcddf@test.com') {
+    throw Error(`Expected to get the key for abcddf@test.com by subkey longid but got ${abcddfBySubkeyLongid.email}`)
+  }
+  if (abcddfBySubkeyLongid.pubkey.id !== '6CF53D2329C2A80828F499D375AA44AB8930F7E9') {
+    throw Error(`Expected to get the key fingerprint 6CF53D2329C2A80828F499D375AA44AB8930F7E9 but got ${abcddfBySubkeyLongid.pubkey.id}`)
+  }
+
+  const [abddefByPrimaryLongid] = await ContactStore.get(undefined, ['5FCC1541CF282951']);
+  if (abddefByPrimaryLongid.email !== 'abddef@test.com') {
+    throw Error(`Expected to get the key for abddef@test.com by primary longid but got ${abddefByPrimaryLongid.email}`)
+  }
+  if (abddefByPrimaryLongid.pubkey.id !== '9E020D9B752FD3FFF17ED9B65FCC1541CF282951') {
+    throw Error(`Expected to get the key fingerprint 9E020D9B752FD3FFF17ED9B65FCC1541CF282951 but got ${abddefByPrimaryLongid.pubkey.id}`)
+  }
+  const [abddefBySubkeyLongid] = await ContactStore.get(undefined, ['EAA7A05FE34F3A1A']);
+  if (abddefBySubkeyLongid.email !== 'abddef@test.com') {
+    throw Error(`Expected to get the key for abddef@test.com by subkey longid but got ${abddefBySubkeyLongid.email}`)
+  }
+  if (abddefBySubkeyLongid.pubkey.id !== '9E020D9B752FD3FFF17ED9B65FCC1541CF282951') {
+    throw Error(`Expected to get the key fingerprint 9E020D9B752FD3FFF17ED9B65FCC1541CF282951 but got ${abddefBySubkeyLongid.pubkey.id}`)
+  }
+  return 'pass';
+})();
+
+BROWSER_UNIT_TEST_NAME(`ContactStore gets a valid pubkey by e-mail, or exact pubkey by longid`);
+(async () => {
+  // Note 1: email differs from pubkey id
+  // Note 2: not necessary to call ContactStore.save, it's possible to always use ContactStore.update
```

Comment:
I wonder if we should stop distinguishing between `save` and `update` and always call it `set`? Do we really currently use them differently, and is there actually a difference?

Reply:
This PR makes no difference between `save` and `update` operations. I left them untouched so we won't need to update a lot of client code in this massive PR.

Reply:
Ok. This would be something to clean up in a followup PR.